packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -134,11 +135,23 @@ export class DatePicker extends BaseComponent<IDatePickerProps, IDatePickerState
     let { formatDate, isRequired, strings, value } = nextProps;
     const errorMessage = (isRequired && !value) ? (strings.isRequiredErrorMessage || '*') : null;
 
-    this.setState({
-      selectedDate: value || new Date(),
-      formattedDate: (formatDate && value) ? formatDate(value) : '',
-      errorMessage: errorMessage
-    });
+    if (errorMessage != null) {
```

Comment:
My thinking was that this error would only occur if there is no value and the only way we can have no value is before the user has selected anything. Granted, in the future, if we add functionality to clear the current selection, we would need to update this.